### PR TITLE
[Dubbo-issue-2131] Check if we can switch serialization/deser between IO and Dubbo Threads. #2131

### PR DESCRIPTION
**Pull request by sinmaystar**
## What is the purpose of the change

Check if we can switch serialization/deser between IO and Dubbo Threads #2131
## Brief changelog

add demo

## Verifying this change

1、dubbo-samples-switch-serialization-thread

From Baiji, Term 268, Team 3, Issue #2131, Q3-3

### CODE REVIEW

File: dubbo-demo-switch-thread/pom.xml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.dubbo</groupId>
+            <artifactId>dubbo-demo-api</artifactId>
+            <version>2.7.0-SNAPSHOT</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.dubbo</groupId>
+            <artifactId>dubbo-config-spring</artifactId>
```
Comment (chickenlj) [inline dubbo-demo-switch-thread/pom.xml]: I think we missed the version here.

**Comment by sinmaystar** [comment]
@chickenlj I have submitted a new sample module and have removed the old demo.

**Comment by chickenlj** [comment]
I am sorry, do we still lack a Consumer? I found `SerializationSwitchThreadProvider.java`, but cannot run this sample without a Consumer.

**Comment by chickenlj** [comment]
@sinmaystar Great job. Thanks, now I will merge this patch to trunk

**Pull request merged**